src/jit_channel/channel_manager.rs

```diff
@@ -0,0 +1,1119 @@
+						self.map_scid_to_peer(scid, counterparty_node_id.clone());
+						let outbound_jit_channel = OutboundJITChannel::new(
+							scid,
+							cltv_expiry_delta,
+							buy_request.payment_size_msat,
+							buy_request.opening_fee_params,
+						);
+
+						peer_state.insert_outbound_channel(scid, outbound_jit_channel);
+
+						let block = scid_utils::block_from_scid(&scid);
+						let tx_index = scid_utils::tx_index_from_scid(&scid);
+						let vout = scid_utils::vout_from_scid(&scid);
+
+						let jit_channel_scid = format!("{}x{}x{}", block, tx_index, vout);
+
+						self.enqueue_response(
+							counterparty_node_id,
+							request_id,
+							Response::Buy(BuyResponse {
+								jit_channel_scid,
+								lsp_cltv_expiry_delta: cltv_expiry_delta,
+								client_trusts_lsp,
+							}),
+						);
+
+						Ok(())
+					}
+					_ => Err(APIError::APIMisuseError {
+						err: format!("No pending buy request for request_id: {:?}", request_id),
+					}),
+				}
+			}
+			None => Err(APIError::APIMisuseError {
+				err: format!("No state for the counterparty exists: {:?}", counterparty_node_id),
+			}),
+		}
+	}
+
+	pub(crate) fn htlc_intercepted(
+		&self, scid: u64, intercept_id: InterceptId, inbound_amount_msat: u64,
+		expected_outbound_amount_msat: u64,
+	) -> Result<(), APIError> {
+		let peer_by_scid = self.peer_by_scid.read().unwrap();
+		if let Some(counterparty_node_id) = peer_by_scid.get(&scid) {
+			let outer_state_lock = self.per_peer_state.read().unwrap();
+			match outer_state_lock.get(counterparty_node_id) {
+				Some(inner_state_lock) => {
+					let mut peer_state = inner_state_lock.lock().unwrap();
+					if let Some(jit_channel) = peer_state.outbound_channels_by_scid.get_mut(&scid) {
+						// TODO: Need to support MPP payments. If payment_amount_msat is known, needs to queue intercepted HTLCs in a map by payment_hash
+						//       LiquidityManager will need to be regularly polled so it can continually check if the payment amount has been received
+						//       and can release the payment or if the channel valid_until has expired and should be failed.
+						//       Can perform check each time HTLC is received and on interval? I guess interval only needs to check expiration as
+						//       we can only reach threshold when htlc is intercepted.
+
+						match jit_channel
+							.htlc_intercepted(expected_outbound_amount_msat, intercept_id)
+						{
+							Ok((opening_fee_msat, amt_to_forward_msat)) => {
+								self.enqueue_event(Event::LSPS2(
+									crate::JITChannelEvent::OpenChannel {
+										their_network_key: counterparty_node_id.clone(),
+										inbound_amount_msat,
+										expected_outbound_amount_msat,
+										amt_to_forward_msat,
+										opening_fee_msat,
+										user_channel_id: scid as u128,
+									},
+								));
+							}
+							Err(e) => {
+								self.channel_manager.fail_intercepted_htlc(intercept_id)?;
+								// remove channel?
```

Comment:
hm i guess technically if it failed because of invalid fee computation the payer could send again and re-use scid?
do we also remove scid from the peer_by_scid map? seems like we'd need to restructure this fn a bit in order to not run into borrow checker issues here and do the removal at the end/outside of these nested if/matches